python/nx-cugraph/nx_cugraph/algorithms/cluster.py

```diff
@@ -56,6 +56,13 @@ def triangles(G, nodes=None):
     return G._nodearrays_to_dict(node_ids, triangles)
 
 
+@triangles._should_run
+def _(G, nodes=None):
+    if nodes is None or nodes not in G:
+        return True
+    return "Fast algorithm when computing for a single node; not worth converting."
```

Comment:
For this and other `should_run`s: does `should_run` get called if the user passes in a backend graph (which does not need converting)?  I remember we honor `backend=` regardless of can/should_run, but is passing a backend graph treated as if a user set `backend=` precedence-wise?

Reply:
`should_run` is only used when the graph needs to be converted. Hence, `should_convert_and_run` would be a more accurate name, but I as recall you liked the shorter name `should_run`. The backend algorithm will always be used if a backend graph if passed or `backend=` argument is used.

Reply:
@eriknw and I talked about this offline.  The use case I was envisioning was one where `should_run` implementations can choose whether or not a backend algorithm is not ideally applied to a particular set of inputs, regardless of if the graph has been converted. For example, if a backend is optimized for very large graphs and the graph is already converted but very small, that backend may choose to let other backends run by returning `False` for `should_run`.  

@eriknw pointed out that a backend graph, in that situation, would have to be converted to another backend graph type (backend-to-backend) or to a NX graph (backend-to-nx) in order to continue, and neither of those are currently supported.  So for now, `should_run` only applies to calls that require a conversion.